tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Query/QueryProjectPropertiesContextTests.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Query
+{
+    public class QueryProjectPropertiesContextTests
+    {
+        [Theory]
+        [MemberData(nameof(ContextsThatAreEqual))]
+        public void Equal(IProjectPropertiesContext a, IProjectPropertiesContext b)
+        {
+            Assert.True(a.Equals(b));
+            Assert.True(a.GetHashCode() == b.GetHashCode());
+        }
+
+        [Theory]
+        [MemberData(nameof(ContextsThatAreNotEqual))]
+        public void NotEqual(IProjectPropertiesContext a, IProjectPropertiesContext b)
+        {
+            Assert.False(a.Equals(b));
+            Assert.False(a.GetHashCode() == b.GetHashCode());
```

Comment:
>a.GetHashCode() == b.GetHashCode() [](start = 25, length = 34)

Hash codes of unequal objects can theoretically collide. Is this always guaranteed to not be the case for IProjectPropertiesContexts?

Reply:
oh, I see you're testing only the ones below so that might not be a problem here.

---
In reply to: [593537232](https://github.com/dotnet/project-system/pull/7006#discussion_r593537232) [](ancestors = 593537232)